main.go

```diff
@@ -10,6 +11,10 @@ func main() {
 	// Run our root CLI command, which contains all underlying command logic and will handle parsing/invocation.
 	err := cmd.Execute()
 
-	// Determine the exit code from any potential error and exit out.
-	os.Exit(exitcodes.GetErrorExitCode(err))
+	// If we have a special (non-zero/success) error code, then exit with it.
+	exitCode := exitcodes.GetErrorExitCode(err)
+	if exitCode != exitcodes.ExitCodeSuccess {
```

Comment:
How would this work if there is a failing test case (exit code 7)? Wouldn't that result in getting that error printed as well? I think the issue that Emilio mentioned with medusa's CLI silently exiting when provided a bad flag is separate?

Reply:
Right no no error bubbled up to main is reflected. So even if an error occurred in cobra, we wouldn't be emitting it.

Reply:
Also good catch, I can unwrap the inner error to make sure it doesn't exist, so we don't log anything (should cover the case of exit code 7)